server/translations.ts

```diff
@@ -141,22 +184,96 @@ function getDocument(filePath) {
     };
   }
 
+  function recordInvalidSourceCommit(
+    isValid: boolean,
+    fileFolder: string,
+    commitHash: string
+  ) {
+    const filePath = "./source-commit-invalid-report.txt";
+    const errorMessage = `- ${commitHash} commit hash is invalid in ${fileFolder}`;
+    if (isValid) return;
+    if (!fs.existsSync(filePath)) {
+      fs.writeFileSync(filePath, "");
+    }
+
+    fs.appendFile(filePath, `${errorMessage}\n`, function (err) {
+      if (err) throw err;
+    });
+  }
+
+  function getCommitBehindFromLatest(
+    fileFolder: string,
+    parentFilePath: string,
+    commitHash: string
+  ): number {
+    try {
+      let count = 0;
+      if (!memCommitStore.has(commitHash)) {
+        execSync(
+          `git log --pretty=format:'%H' --name-only ${commitHash}..${memCommitStoreOldest}`,
+          {
+            cwd: CONTENT_ROOT,
+          }
+        )
+          .toString()
+          .trimEnd()
+          .split("\n\n")
+          .forEach((commit) => {
+            const [hash, ...files] = commit.split("\n");
+            memCommitStore.set(hash, files);
+          });
+        memCommitStoreOldest = commitHash;
+      }
+      for (const [hash, files] of memCommitStore.entries()) {
+        if (hash === commitHash) {
+          recordInvalidSourceCommit(
+            files.includes(parentFilePath),
+            fileFolder,
+            commitHash
+          );
```

Comment:
It took me a moment to figure out why we were recording an invalid commit when the hashes match - I think this is a bit clearer with a wrapping `if (files.includes(parentFilePath)`, which is what I've done in my commit.

Reply:
Reflected in the commit below. Details will be explained in the full comment.

- https://github.com/mdn/yari/pull/8338/commits/afcee51449565915cbb2faf67fe30ed1b05465ad

Reply:
I agree. The first intention was to get rid of it because I thought nesting of if statements was not good, but it seems that there was a lack of clarity.